src/main/java/org/jabref/logic/cleanup/PageFieldCleanup.java

```diff
@@ -0,0 +1,34 @@
+package org.jabref.logic.cleanup;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.jabref.model.FieldChange;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.StandardField;
+
+/**
+ * adds the article ID of a journal as the page count, but only if the page field is empty
+ */
+public class PageFieldCleanup implements CleanupJob {
+
+    @Override
+    public List<FieldChange> cleanup(BibEntry entry) {
+        List<FieldChange> changes = new ArrayList<>();
+        Optional<String> doiAsString = entry.getField(StandardField.DOI);
+
+        if (doiAsString.isPresent() && !entry.hasField(StandardField.PAGES)) {
+            String articleId = new String();
+            int index = doiAsString.get().length() - 1;
+            while (Character.isDigit(doiAsString.get().charAt(index))) {
```

Comment:
Since this issue only concerns articles from Physical Review, I would suggest to use a regex based on the format outlined in https://github.com/JabRef/jabref/issues/7019#issuecomment-711036239. In particular, make sure it only applies to doi's to Physical Review, and not all dois ending on some number.